File: frontend/csc13001-plant-pos/csc13001-plant-pos/Model/InventoryItem.cs
using System.Text.Json.Serialization;

namespace csc13001_plant_pos.Model
{
    public class InventoryItem
    {
        [JsonPropertyName("inventoryItemId")]
        public int InventoryItemId { get; set; }
        [JsonPropertyName("product")]
        public Product Product { get; set; }
        [JsonPropertyName("purchasePrice")]
        public decimal PurchasePrice { get; set; }
        [JsonPropertyName("quantity")]
        public int Quantity { get; set; }
        [JsonPropertyName("remainingQuantity")]
        public int RemainingQuantity { get; set; }
        public decimal TotalItemPrice => Quantity * PurchasePrice;
    }
}

File: backend/src/main/java/csc13001/plantpos/inventory/models/InventoryItem.java
package csc13001.plantpos.inventory.models;

import csc13001.plantpos.product.Product;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonSetter;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "inventory_items")
public class InventoryItem {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "inventory_item_id")
    private Long inventoryItemId;

    @JsonIgnore
    @ManyToOne
    @JoinColumn(name = "inventory_id", nullable = false)
    private Inventory inventory;

    @ManyToOne
    @JoinColumn(name = "product_id", nullable = false)
    private Product product;

    @Column(name = "purchase_price", nullable = false)
    private BigDecimal purchasePrice;

    @Column(name = "quantity", nullable = false)
    private Integer quantity;

    @Column(name = "remaining_quantity", nullable = false)
    private Integer remainingQuantity;

    @JsonSetter("productId")
    public void setProductId(Long productId) {
        this.product = Product.builder().productId(productId).build();
    }
}